readthedocs/rtd_tests/tests/test_views.py

```diff
@@ -241,3 +241,30 @@ def test_project_admins_can_delete_subprojects_that_they_are_not_admin_of(self):
             '/dashboard/my-mainproject/subprojects/my-subproject/delete/')
         self.assertEqual(response.status_code, 302)
         self.assertTrue(self.subproject not in [r.child for r in self.project.subprojects.all()])
+
+class URLResolution(TestCase):
+    def setUp(self):
+        self.user1 = new(User, username='foo+bar')
+        self.user1.set_password('test1')
+        self.user1.save()
+
+        self.user2 = new(User, username='abc+def@ghi.jkl')
+        self.user2.set_password('test2')
+        self.user2.save()
+
+        self.user3 = new(User, username='abc-def+ghi')
+        self.user3.set_password('test3')
+        self.user3.save()
+
+        self.client.login(username='foo+bar', password='test1')
+
+    def test_profile_details_page(self):
+
+        response = self.client.get('/profiles/foo+bar/')
```

Comment:
Instead of creating these users and accessing the page itself, isn't it better to use the resolver and check that it matches?

Reply:
Hi @humitos, Sorry for the delay on this. I was busy with class tests in my college. 
Can you please elaborate on this as I am a bit confused about how to proceed.

Reply:
Sorry, I meant the `reverse` function from django:

```
In [1]: from django.core.urlresolvers import reverse

In [3]: reverse('profiles_profile_detail', kwargs={'username': 'foo+bar'})
Out[3]: '/profiles/foobar/'
```

and check the URL returned with `assertEqual`.

Reply:
Thanks for the explanation @humitos, I have made the changes